lib/fields.c

```diff
@@ -21,7 +21,7 @@
  *
  * The supplied field is scanned for non-printable and other illegal
  * characters.
- *  + -1 is returned if an illegal character is present.
+ *  + -1 is returned if an illegal or control character is present.
  *  +  1 is returned if no illegal characters are present, but the field
```

Comment:
I'm going to be a bit picky here. Are control characters considered non-printable characters? If not, can you make sure to update the comment for return value `1`?